src/codegen/sys/tests.rs

```diff
@@ -213,20 +213,45 @@ fn generate_layout_c(
     writeln!(w)?;
     writeln!(w, "#include \"manual.h\"")?;
     writeln!(w, "#include <stdalign.h>")?;
-    writeln!(w, "#include <stdio.h>")?;
-    writeln!(w)?;
-    writeln!(w, "{}", r"int main() {")?;
+    writeln!(
+        w,
+        "{}",
+        r####"
+typedef struct {
+    const char *name;
+    size_t size;
+    size_t alignent;
+} Layout;
 
-    for ctype in ctypes {
-        writeln!(
+const Layout LAYOUTS[] = {"####
+    )?;
+
+    let last = ctypes.len() - 1;
+    for (i, ctype) in ctypes.iter().enumerate() {
+        write!(w, "{}", "    { ")?;
+        write!(
             w,
-            "    printf(\"%s;%zu;%zu\\n\", \"{ctype}\", sizeof({ctype}), alignof({ctype}));",
+            "\"{ctype}\", sizeof({ctype}), alignof({ctype})",
             ctype = ctype.name
         )?;
+
+        if i == last {
+            writeln!(w, "{}", " }")?;
+        } else {
+            writeln!(w, "{}", " },")?;
+        }
     }
 
-    writeln!(w, "    return 0;")?;
-    writeln!(w, "{}", r"}")
+    writeln!(
+        w,
+        "{}",
+        r####"};
+
+const Layout *c_layouts(size_t *n) {
+    *n = sizeof(LAYOUTS) / sizeof(Layout);
```

Comment:
```suggestion
    *n = G_N_ELEMENTS(LAYOUTS);
```

Reply:
This file does not include glib... I would keep it minimal C, but no strong opinion

Reply:
The other C code uses `g_strdup()` instead of `strdup()`, etc., that's why I was suggesting to use glib API here. Your call :)

Reply:
yeah... in the other test I had to pull in glib, because C without `g_strdup_printf` is a nightmare :)
While it is true that we are pulling in glib, I would still keep this file as minimal as possible.